src/mrinufft/operators/interfaces/nudft_numpy.py

```diff
@@ -4,23 +4,55 @@
 
 import numpy as np
 import scipy as sp
-
 from ..base import FourierOperatorCPU
-from mrinufft._utils import proper_trajectory
+from mrinufft._utils import proper_trajectory, get_array_module
 
 
 def get_fourier_matrix(ktraj, shape, dtype=np.complex64, normalize=False):
-    """Get the NDFT Fourier Matrix."""
+    """Get the NDFT Fourier Matrix.
+
+    Parameters
+    ----------
+    ktraj : array_like
+        The k-space coordinates for the Fourier transformation.
+    shape : tuple of int
+        The dimensions of the output Fourier matrix.
+    dtype : data-type, optional
+        The data type of the Fourier matrix, default is np.complex64.
+    normalize : bool, optional
+        If True, normalizes the matrix to maintain numerical stability.
+
+    Returns
+    -------
+    matrix
+        The NDFT Fourier Matrix.
+    """
+    xp = get_array_module(ktraj)
     ktraj = proper_trajectory(ktraj, normalize="unit")
     n = np.prod(shape)
     ndim = len(shape)
-    matrix = np.zeros((len(ktraj), n), dtype=dtype)
-    r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
-    grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
-    traj_grid = ktraj @ grid_r
-    matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
+    dtype = xp.complex64
+    device = getattr(ktraj, "device", None)
+
+    r = [xp.linspace(-s / 2, s / 2 - 1, s) for s in shape]
+    if xp.__name__ == "torch":
+        r = [x.to(device) for x in r]
+
+    grid_r = xp.meshgrid(*r, indexing="ij")
+    grid_r = xp.reshape(xp.stack(grid_r), (ndim, n))
+    traj_grid = xp.matmul(ktraj, grid_r)
+    matrix = (
+        xp.exp(-2j * xp.pi * traj_grid).to(dtype).to(device).clone()
+        if xp.__name__ == "torch"
+        else (xp.exp(-2j * xp.pi * traj_grid, dtype=dtype))
+    )
```

Comment:
```suggestion
    matrix = xp.exp(-2j * xp.pi * traj_grid)
    if xp.__name__ == "torch":
        matrix.to(dtype=dtype, device=device, copy=True)
    )
```

Reply:
I think I have to use matrix = matrix.to(...) here, otherwise it will occur an error in the autograd function of pytorch.